Add release notes for v24.1.0-alpha.2 (#18353)
cockroachdb/docs: src/current/_includes/releases/v24.1/v24.1.0-alpha.2.md

## v24.1.0-alpha.2

Release Date: March 11, 2024

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v24-1-0-alpha-2-security-updates">Security updates</h3>

- [DB Console](https://www.cockroachlabs.com/docs/v23.2/ui-overview) `session` cookie is now marked `HttpOnly` to prevent it from being read by any JavaScript code. [#119261][#119261]

> - [DB Console](https://www.cockroachlabs.com/docs/v23.2/ui-overview) cookies are marked `Secure` for the browser when the cluster is running in secure mode. [#119261][#119261]

Author: ✅

<h3 id="v24-1-0-alpha-2-general-changes">General changes</h3>

- [Gateways](https://www.cockroachlabs.com/docs/v23.2/architecture/life-of-a-distributed-transaction#gateway) will now detect faulty or stalled [replicas](https://www.cockroachlabs.com/docs/v23.2/architecture/overview#architecture-replica) and use other replicas instead, which can prevent them getting stuck in certain cases (e.g. with [disk stalls](https://www.cockroachlabs.com/docs/v23.2/cluster-setup-troubleshooting#disk-stalls)). This behavior can be disabled via the [cluster setting](https://www.cockroachlabs.com/docs/v23.2/cluster-settings) `kv.dist_sender.circuit_breaker.enabled`. [#118943][#118943]

<h3 id="v24-1-0-alpha-2-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>

> - Added a new `SUBJECT` [role](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#roles) option. This role option can be set to a subject distinguished name in RFC2253 or RFC4514 format. If set, then during client certificate authentication, certs that do not match the configured distinguished name will be rejected. [#119135][#119135]

Author: ✅

> - [Changefeeds](https://www.cockroachlabs.com/docs/v23.2/create-changefeed) support a new scheme `azure-event-hub://` for Kafka data streaming to Azure event hubs. The `sinkURL` must include mandatory parameters `shared_access_key_name` and `shared_access_key`. By default and by requirements, the options `tls_enabled=true`, `sasl_handshake=true`, `sasl_enabled=true`, and `sasl_mechanism=PLAIN` are applied, as they are the only supported options. Other parameters such as `topic_name` and `topic_prefix` are also supported. An example URI is: `azure-event-hub://myeventhubs.servicebus.windows.net:9093?shared_access_key_name=abc&shared_access_key=123`. [#115806][#115806]

Reviewer: ```suggestion
- [Changefeeds](https://www.cockroachlabs.com/docs/v23.2/create-changefeed) support a new scheme `azure-event-hub://` for Kafka data streaming to Azure event hubs. The `sinkURL` must include mandatory parameters `shared_access_key_name` and `shared_access_key`. By default and as required, the options `tls_enabled=true`, `sasl_handshake=true`, `sasl_enabled=true`, and `sasl_mechanism=PLAIN` are applied, as they are the only supported options. Other parameters such as `topic_name` and `topic_prefix` are also supported. An example URI is: `azure-event-hub://myeventhubs.servicebus.windows.net:9093?shared_access_key_name=abc&shared_access_key=123`. [#115806][#115806]
```

Author: ✅

<h3 id="v24-1-0-alpha-2-sql-language-changes">SQL language changes</h3>

> - Added an option for node-level [sequence](https://www.cockroachlabs.com/docs/v23.2/serial) caching. All the sessions on the node can share the same cache, which can be concurrently accessed. The `serial_normalization` [session variable](https://www.cockroachlabs.com/docs/v23.2/set-vars#supported-variables) can now be set to the value `sql_sequence_cached_node`. If this value is set, the [cluster setting](https://www.cockroachlabs.com/docs/v23.2/cluster-settings) `sql.defaults.serial_sequences_cache_size` can be used to control the number of values to cache in a node, with a default of 256. The `PER NODE CACHE` sequence option (syntax is `[ [ PER NODE ] CACHE # ]`)  is now fully implemented and will allow nodes to cache sequence numbers. A cache size of 1 means there is no cache, and cache sizes of less than 1 are not valid. [#118546][#118546]

Reviewer: Totally optional: This might be easier to follow if the session/cluster var info were in a bullet and the SQL option in another.

Author: I think I will punt on this for now.  my brain isn't working that well today (DST sleep) and I don't want to introduce an error in rephrasing

<h3 id="v24-1-0-alpha-2-bug-fixes">Bug fixes</h3>

- Fixed a bug that prevented the use of [PL/pgSQL](https://www.cockroachlabs.com/docs/v23.2/plpgsql) routines with complex variable names that require double quotes. This bug had existed since v23.2. [#119034][#119034]
- Fixed a bug that could cause creation of a syntactically invalid [PL/pgSQL](https://www.cockroachlabs.com/docs/v23.2/plpgsql) routine to return the wrong error. This bug had existed since v23.2. [#119034][#119034]
- Fixed a bug that could result in a syntax error if a [PL/pgSQL](https://www.cockroachlabs.com/docs/v23.2/plpgsql) routine was created with an escaped string constant in the routine body. This bug had existed since v23.2. [#119034][#119034]

> - Fix a bug where running [`RESTORE`](https://www.cockroachlabs.com/docs/v23.2/restore) on certain backups would open a very large number of connections to the backup storage provider. [#119840][#119840]

Reviewer: ```suggestion
- Fixed a bug where running [`RESTORE`](https://www.cockroachlabs.com/docs/v23.2/restore) on certain backups would open a very large number of connections to the backup storage provider. [#119840][#119840]
```

Author: ✅

> - If a user has the `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#managing-privileges), then they should not be able to see constants inside of queries that originate from other [users](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#roles) in the [`SHOW SESSIONS`](https://www.cockroachlabs.com/docs/v23.2/show-sessions) output. Previously, this redaction was not occurring, but this is now fixed. [#119820][#119820]

Reviewer: ```suggestion
- Previously, a user with the `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#managing-privileges) could see constants inside of queries that originate from other [users](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#roles) in the [`SHOW SESSIONS`](https://www.cockroachlabs.com/docs/v23.2/show-sessions) output. This information is now properly redacted. [#119820][#119820]
```

Author: ✅

> - The [`SHOW QUERIES`](https://www.cockroachlabs.com/docs/v23.2/show-queries) and [`SHOW STATEMENTS`](https://www.cockroachlabs.com/docs/v23.2/show-statements) commands were incorrectly requiring that the user had the `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#managing-privileges). However, a user should always be able to view their own queries, even without this privilege. This is now fixed. [#119820][#119820]

Reviewer: ```suggestion
- Previously, the [`SHOW QUERIES`](https://www.cockroachlabs.com/docs/v23.2/show-queries) and [`SHOW STATEMENTS`](https://www.cockroachlabs.com/docs/v23.2/show-statements) commands incorrectly required the user to have the `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.2/security-reference/authorization#managing-privileges). This is now fixed, as a user should always be able to view their own queries. [#119820][#119820]
```

Author: ✅  (kept the phrase "even without this privilege" since I think it makes it even clearer)
